cli/options.json

```diff
@@ -222,6 +219,9 @@
       "",
       " mutable-globals     Mutable global imports and exports.",
       " sign-extension      Sign-extension operations",
+      " nontrapping-f2i     Non-trapping float to integer ops.",
+      " bulk-memory         Bulk memory operations.",
+      " simd                SIMD types and operations.",
```

Comment:
It's too early. SIMD doesn't support on Safari. Bulk memory generate suboptimal code on binaryen yet. Also has suboptimal performance: https://bugs.chromium.org/p/v8/issues/detail?id=11775&q=maxgraey&can=2

`nontrapping-f2i` I enabled here: https://github.com/AssemblyScript/assemblyscript/pull/2107

Reply:
Also not 100% sure about SIMD given Safari, but seems ok-ish given that it is standardized and not used by most projects anyhow. Regarding Bulk Memory, we are only using that for memset/memmove currently - how would it lead to problems?

Reply:
Regarding #2107, let's rebase the specific implementation tweaks on this branch :)

Reply:
> but seems ok-ish given that it is standardized and not used by most projects anyhow.

SIMD uses during bulk memory lowering for constant size equal `8`. Also I would like add some fast simd path in std. Also note SIMD banned in smart contracts (even integer ops)

Reply:
> Regarding Bulk Memory, we are only using that for memset/memmove currently - how would it lead to problems?

It increase size and reduce performance in some trivial cases like `memory.fill(p, 0, 3)` or `memory.fill(p, 0, 0)`. Before enabling it better merge some PRs in binaryen and wait better codegen on v8

Reply:
Wondering, are the Bulk Memory issues perhaps related to lowering it, and not so much to when it is enabled?

Reply:
> Regarding #2107, let's rebase the specific implementation tweaks on this branch :)

I would like keep that PR separate if you don't mind. This will be easier to find and easier to undo if something goes wrong. I suggest just merge that PR immediate after this PR

Reply:
> How would the picture change when enabling Bulk Memory but not SIMD?

I suggest you consider including bulk memory later. With a preliminary study of the pros and cons.

Reply:
I am considering Bulk Memory for its significant code size advantages. We can talk about SIMD, but so far I am not seeing a good reason for not enabling BM.

Reply:
> I am considering Bulk Memory for its significant code size advantages. We can talk about SIMD, but so far I am not seeing a good reason for not enabling BM.

The problem is that for trivial cases the code slows down. That is, it becomes more expensive to create classes and small objects. I'm not sure yet that bulk memory is ready to be enabled by default. Tests are needed

Reply:
Testing with the compiler I am seeing a fluctuating 3-4% difference or so, typically in `parse` stage. Hard to tell if that's merely noise, comes from the instruction itself or from what Binaryen does or does not do. In general I think that we should enable new features and improve upon them incrementally, which also has a higher chance to happen once it's out in the wild.

Reply:
I now saw in the fixtures however that Bulk Memory is lowed to SIMD, as you said, so I've put SIMD behind a flag again.